test/e2e/storage/testsuites/provisioning.go

```diff
@@ -440,6 +443,231 @@ func (p *provisioningTestSuite) DefineTests(driver storageframework.TestDriver,
 		l.testCase.TestDynamicProvisioning(ctx)
 	})
 
+	ginkgo.It("should provision storage with pvc data source for readonly filesystem", func(ctx context.Context) {
+		if !dInfo.Capabilities[storageframework.CapPVCDataSource] {
+			e2eskipper.Skipf("Driver %q does not support cloning - skipping", dInfo.Name)
+		}
+
+		init(ctx)
+
+		if l.config.ClientNodeSelection.Name == "" {
+			// Schedule all pods to the same topology segment (e.g. a cloud availability zone), some
+			// drivers don't support cloning across them.
+			if err := ensureTopologyRequirements(ctx, &l.config.ClientNodeSelection, l.cs, dInfo, 1); err != nil {
+				framework.Failf("Error setting topology requirements: %v", err)
+			}
+		}
+		testConfig := storageframework.ConvertTestConfig(l.config)
+		expectedContent := fmt.Sprintf("Hello from namespace %s", f.Namespace.Name)
+
+		l.sourcePVC.Name = "source-pvc"
+		dataSourceRef := preparePVCDataSourceForProvisioning(ctx, f, testConfig, l.cs, l.sourcePVC, l.sc, pattern.VolMode, expectedContent)
+		l.pvc.Spec.DataSourceRef = dataSourceRef
+		l.testCase.NodeSelection = testConfig.ClientNodeSelection
+
+		// Cloning fails if the source disk is still in the process of detaching, so we wait for the VolumeAttachment to be removed before cloning.
+		volumeAttachment := e2evolume.GetVolumeAttachmentName(ctx, f.ClientSet, testConfig, l.testCase.Provisioner, dataSourceRef.Name, l.sourcePVC.Namespace)
+		e2evolume.WaitForVolumeAttachmentTerminated(ctx, volumeAttachment, f.ClientSet, f.Timeouts.DataSourceProvision)
+
+		originalClaim, err := l.testCase.Client.CoreV1().PersistentVolumeClaims(l.pvc.Namespace).Get(ctx, l.sourcePVC.Name, metav1.GetOptions{})
+		framework.ExpectNoError(err, "Failed to get pvc: %v", err)
+
+		var originFSSize int
+		if pattern.VolMode != "Block" {
+			// Create pod to get mount path of fs
+			createdClaims := []*v1.PersistentVolumeClaim{originalClaim}
+			pod, err := e2epod.CreatePod(ctx, l.testCase.Client, f.Namespace.Name, nil, createdClaims, true, "", true)
+			framework.ExpectNoError(err, "Failed to create pod: %v", err)
+			mountpath := findVolumeMountPath(pod, originalClaim)
+			gomega.Expect(mountpath).ShouldNot(gomega.BeEmpty())
+
+			// Save filesystem size of the origin volume.
+			originFSSize, err = getFilesystemSizeBytes(pod, mountpath)
+			framework.ExpectNoError(err, "Failed to obtain filesystem size of a volume mount: %v", err)
+		}
+		originalSize := originalClaim.Status.Capacity.Storage()
+		biggerSize := originalSize.DeepCopy()
+		biggerSize.Add(*resource.NewQuantity(1, resource.BinarySI))
+
+		// Create claim w/ bigger size than original
+		biggerROClaim := l.testCase.Claim.DeepCopy()
+		biggerROClaim.Name = "bigger-claim"
+		biggerROClaim.Spec.Resources = v1.ResourceRequirements{
+			Requests: v1.ResourceList{
+				v1.ResourceStorage: biggerSize,
+			},
+		}
+		biggerROClaim.Spec.AccessModes = []v1.PersistentVolumeAccessMode{
+			v1.PersistentVolumeAccessMode(v1.ReadOnlyMany),
```

Comment:
Not all volume types support ROX access mode. You may need to guard these test cases with a new test capability.

Reply:
Added CapReadOnlyMany, good idea thanks! Do I need to go and opt some drivers out or is the onus on driver owners to do this if their k8s integration tests start to fail?

Reply:
Ah it looks like Capabilities are opt-in so opting in PD in this PR is enough I think? lmk if I'm missing something though